model/task/db.go

```diff
@@ -788,25 +788,36 @@ func needsContainerAllocation() bson.M {
 //     have been met.
 //  2. The container is allocated but the agent has not picked up the task yet.
 func IsContainerTaskScheduledQuery() bson.M {
+	query := IsUndispatchedContainerTasksQuery()
+	query["$or"] = dependenciesAreMetQuery()
+	return query
+}
+
+// IsUndispatchedContainerTasksQuery returns a query returning all tasks that
```

Comment:
comment nit: "returning all undispatched container tasks" has less redundancy 
function title nit ( that also applies to the existing query above so my bad): functions starting with "Is" usually imply that we're returning a boolean. Really it's the ScheduledContainerTasksQuery and the UndispatchedContainerTasksQuery.